jenkins/e2e-image/e2e-runner.sh

```diff
@@ -273,25 +327,47 @@ EXIT_CODE=0
 # We get the Kubernetes tarballs unless we are going to use old ones
 if [[ "${JENKINS_USE_EXISTING_BINARIES:-}" =~ ^[yY]$ ]]; then
     echo "Using existing binaries; not cleaning, fetching, or unpacking new ones."
-elif [[ "${KUBE_RUN_FROM_OUTPUT:-}" =~ ^[yY]$ ]]; then
-    # TODO(spxtr) This should probably be JENKINS_USE_BINARIES_FROM_OUTPUT or
-    # something, rather than being prepended with KUBE, since it's sort of a
-    # meta-thing.
-    clean_binaries
-    fetch_output_tars
-elif [[ "${JENKINS_USE_SERVER_VERSION:-}" =~ ^[yY]$ ]]; then
-    # This is for test, staging, and prod jobs on GKE, where we want to
-    # test what's running in GKE by default rather than some CI build.
-    clean_binaries
-    fetch_server_version_tars
-elif [[ "${JENKINS_USE_GCI_VERSION:-}" =~ ^[yY]$ ]]; then
-    clean_binaries
-    fetch_gci_version_tars
 else
-    # use JENKINS_PUBLISHED_VERSION, default to 'ci/latest', since that's
-    # usually what we're testing.
     clean_binaries
```

Comment:
I get that this eliminates a couple duplicate lines of clean_binaries, but I do not think the additional branch depth justifies this. Especially since it makes it harder to understand what you are adding.